src/rat/core/run_sarea.py

```diff
@@ -40,11 +41,20 @@ def run_sarea(start_date, end_date, datadir, reservoirs_shpfile, shpfile_column_
             log.exception(f"Surface area calculation failed for {reservoir_name}.")
             no_failed_files += 1
     if no_failed_files:
-        log_level1.warning(f"Surface area was not calculated for {no_failed_files} reservoirs.")
+        log_level1.warning(f"Surface area was not calculated for {no_failed_files} reservoirs.")     
     if Optical_files:
         log_level1.warning(f"Surface area was calculated using only Optical data and not TMS-OS for {Optical_files} reservoirs. It can be due to insufficient SAR data. Please refer level-2 log file for more details.")
     if Partial_optical_tmsos_files:
-        log_level1.warning(f"Surface area was calculated partially using only Optical data and rest using TMS-OS for {Partial_optical_tmsos_files} reservoirs. It can be due to more Optical data than SAR data. Please refer level-2 log file for more details.") 
+        log_level1.warning(f"Surface area was calculated partially using only Optical data and rest using TMS-OS for {Partial_optical_tmsos_files} reservoirs. It can be due to more Optical data than SAR data. Please refer level-2 log file for more details.")
+        
+    #Running Bot Filter
+    bot_thresholds = [filt_options['bias_threshold'],filt_options['outlier_threshold'],filt_options['trend_threshold']]
+    if(min(bot_thresholds)<0 or max(bot_thresholds) > 9):
+        if(filt_options['apply']==True):
+            log_level1.error(f"BOT Filter run failed for {reservoir_no} reservoirs")
```

Comment:
reservoir_no is referenced outside the for loop. You can say failed for all reservoirs or use the total no of reservoirs.

Reply:
Removed in new commit